### DISCUSSION TITLE
MissionControl 0.6.0

### DISCUSSION THREAD

**Discussion by ndeadly** [ndeadly/MissionControl#356] [General]
<p align="left">
<img alt="GitHub Releases" src="https://img.shields.io/github/downloads/ndeadly/MissionControl/v0.6.0/total">
</p>

This is the ninth official release of Mission Control

This release brings full support for firmware 13.0.0, along with a few minor additions and bugfixes. See the changelog below for details.

New users landing on this release page should first check out the readme on the [main project page](https://github.com/ndeadly/MissionControl) for the official project documentation. There you can find installation and usage instructions along with an FAQ section that will answer most of your questions.

_Note: 13.0.0 firmware made changes to the system to facilitate bluetooth audio support. While I have updated Mission Control to support the changes that concern hid devices, it's possible that there are issues that could arise from untested scenarios involving audio devices. If you notice any strange behaviour with audio devices that only occurs with mission control installed, please open an issue on the issue tracker._

***Atmosphère >= 1.1.0 is required to run this release.***

### Important notice for users of Xbox One controllers
With [news](https://news.xbox.com/en-us/2021/09/08/xbox-controller-firmware-update-rolling-out-to-insiders-starting-today/) of a new insiders firmware update to bring next-gen features to existing Xbox One controllers, it may be tempting to download and install the firmware when it becomes available to you. From what I can tell from the linked news article and some intial reports, this update _may_ configure your controller to use Bluetooth LE for connections instead of the current classic Bluetooth, effectively converting it into a Series S/X controller. __This means your controller will likely be rendered incompatible with Mission Control until I am able to add Bluetooth LE support. You have been warned.__

### Changelog:
- Added full support for 13.0.0 firmware.
- Added bluetooth ips patches for 13.0.0.
- The `/config/MissionControl/controllers` directory is now created as required if not present. This avoids potential crashes eg. if the user has deleted this directory, or has failed to install the module correctly.
- Fixed potential incorrect battery level and report charging status for Wii U Pro Controller. (Thanks @GaryOderNichts)
- External power/USB status is now set for controllers known to report it.
- Xbox One controllers now report full battery when powered via USB, instead of the zero value reported by the controller.

<hr /><em>This discussion was created from the release <a href='https://github.com/ndeadly/MissionControl/releases/tag/v0.6.0'>MissionControl 0.6.0</a>.</em>

**Comment by mighty-abo** [comment]
I have updated to  MissionControl 0.6.0 and 13.0.0 firmware ,but here comes a new problem  to me.While pairing my controller lanShen X1Pro  requested @#213,the screen shows connetting success,but the led just flash 1s and disconnected.If there is a bug or not,hope can
fix it.

**Reply by ndeadly** [reply]
Which version are you updating from? In 0.5.0 I changed the default behaviour of blocking the SetTsi command to all controllers as it was responsible for some latency and network bandwidth issues. It's likely this is another controller that doesn't like receiving this command and will need to have it blocked again.

If you have been previously using the controller with 0.5.x successfully then it might be some new 13.0.0 behaviour that the controller doesn't like and will require some investigation.

**Comment by bibliophilical** [comment]
With this release (based on the changelog), it means that Wii U Pro Con will have a correct battery level report?

**Reply by ndeadly** [reply]
I was under the impression it was already correct, but @GaryOderNichts was not so sure. What this does is reads the battery level (along with charging status) from the button input reports instead of the periodic status reports used by other Wii controllers. This isn't well documented and I wasn't aware that the Wii U Pro controller sent this info previously.

**Comment by mighty-abo** [comment]
Update from 0.4.1.Maybe it can't work in 0.5.0,so sad

**Reply by ndeadly** [reply]
You misunderstand. If you skipped 0.5.x then it's almost certainly the issue I mentioned above and just requires a new build that blocks sending the command to your controller

**Reply by Mark7458** [reply]
I have an issue installing it on my Switch. The console doesn't see my ps4 controller while in pairing mode. I have recently updated Atmosphere to the latest version (1.1.1), may this be a problem? 
I had no previous version of mission control installed. Maybe there are some important files missing in this version but included in the previous one?

**Reply by mighty-abo** [reply]
Got that.So what should I do now？

**Reply by ndeadly** [reply]
@Mark7458 did you follow the [usage](https://github.com/ndeadly/MissionControl#usage) and pairing instructions in the readme? (ie. rebooted the console after installation, enter the `Change Grip/Order` screen to pair the controller etc.)

If you've followed the instructions correctly and it's still not pairing, try hitting the reset button on the back of the controller and attempt to pair again, restarting the sync mode several times if necessary. It should eventually pair to the console. 

Note that some counterfeit DS4 controllers are not seen by the console at all and can not be paired, likely due to an incompatible bluetooth chipset.

**Reply by ndeadly** [reply]
@mighty-abo try this

[MissionControl-0.6.0-lanshen-disable-settsi-67c14d5.zip](https://github.com/ndeadly/MissionControl/files/7229641/MissionControl-0.6.0-lanshen-disable-settsi-67c14d5.zip)

**Reply by Mark7458** [reply]
Thanks for the answer. 
Yes, all done but it can't be detected in bluetooth mode (but DS4 connects via usb using sys-con).
I'll try with a different controller and see what happens. 
I'll keep you updated.

**Reply by mighty-abo** [reply]
Bravo!It works.

**Reply by Mark7458** [reply]
@ndeadly I tried also with the Xbox controller, same problem. I've done exactly what explained in the guide. 
I can check it in Hekate Toolbox so the files are actually in the SD. No error during the copy/paste. 

Can you verify if the are some compatibility issues with Atmosphere v1.1.1?

**Reply by Mark7458** [reply]
> If you've followed the instructions correctly and it's still not pairing, try hitting the reset button on the back of the controller and attempt to pair again, restarting the sync mode several times if necessary. It should eventually pair to the console

@ndeadly I skipped that part... 
Done it and it works now. I didn't know it's necessary every time you pair the controller with a different console, but makes sense actually.
Sorry for bothering you.

**Comment by impeeza** [comment]
English please

**Comment by J0k3rVN** [comment]
Pls! Help me! I cant install mission control, i will copy all file and paste in my SD card but not work. My atmosphere is Kefir 621

**Reply by ndeadly** [reply]
Did you follow the [installation](https://github.com/ndeadly/MissionControl#installation) and pairing instructions?

Not working can mean many things. You're going to have to be more specific if you want help.

**Reply by J0k3rVN** [reply]
i will paste all file in my SD card but in my Hbmenu dont have mission cotrol, i can install with HB app store but in HB app store have 0.5.0. (sorry my english very bad)

**Reply by impeeza** [reply]
Have you read the readme and instructions?  MissionControl by now hasn't an interface, mission control intercept the Bluetooth communication, the user interface is the same of the system.

There is an application but is only intented for diagnostics and isn't part of the official release

**Comment by impeeza** [comment]
Is supposed what Kefir 621 is a pre-cooked package with MissionControl 0.6.0 preinstalled, so is supposed mission controll should work fine on your environment IF YOU follor all the instructions, the Kefir and MissionControl ones,  please repeat the installation and read the use instructions stated on main page of MissionControll.

**Comment by diegolc88** [comment]
I’ve just updated to 13.0.0 and atm 1.1.1|S and Mission Contoller 0.6.0. I’ve done it in other versions, it always works, but in this version no. “Panic occurred while runing atmosphere” when conect PS4 controller
![Uploading 9AAAB80F-0EB1-4268-963B-38F5CD6D8D37.jpeg…]()
, I need help
Thanks
![9B23BB68-6008-44A8-86A0-9231BF30297F](https://user-images.githubusercontent.com/93420452/139522137-f036babd-e256-4fd2-b879-73dbd5d26431.jpeg)

**Reply by ndeadly** [reply]
I'm going to guess there's some kind of version mismatch going on here. Why not use the latest versions of both Atmosphere (1.2.2) and Mission Control (0.6.1)?

**Reply by diegolc88** [reply]
Atmosphere 1.2.2 works with 13.0.0, no? I change the sigpages and the folder atmosphere, no?
Thanks ndeadly, I need the ps4 controller woks in my switch…

**Reply by diegolc88** [reply]
13.1.0 and Atmosphere 1.2.1, it using Mission Control 0.6.1, it doesn’t work… same error, any ideas.
I’ve got atmosphere/exefs_patches/es_patches in my SD:
Is it correct?
Thanks
![9B4E576D-2ACC-4E70-9AEF-CD1188DC433E](https://user-images.githubusercontent.com/93420452/139543029-f52da46b-8528-421b-86be-ad8df4d519b1.jpeg)

**Reply by diegolc88** [reply]
It works in atm 1.2.2
Thanks you
I love you, ndeadly
